src/store/actions/authActions/socialLoginAction.js

```diff
@@ -0,0 +1,75 @@
+import firebase from 'firebase';
+import { SOCIAL_LOGIN_SUCCESS, SOCIAL_LOGIN_FAILURE } from '../actionTypes';
+import ShowModal from '../changeFormAction';
+import axiosConfig from '../../../axiosConfig';
+
+
+export const createLoginSuccess = () => ({
+  type: SOCIAL_LOGIN_SUCCESS,
+});
+
+export const createLoginFailure = error => ({
+  type: SOCIAL_LOGIN_FAILURE,
+  error,
+});
+
+export const socialLoginAction = (accessToken, authProvider, accessTokenSecret = '') => (dispatch) => {
+  let provider = authProvider;
+  // google provider is named differently in our backend.
+  if (authProvider === 'google') {
+    provider = 'google-oauth2';
+  }
+
+  const authData = {
+    access_token: accessToken,
+    access_token_secret: accessTokenSecret,
+    provider,
+  };
+
+  return axiosConfig.request({
+    method: 'post',
+    data: authData,
+    url: '/api/user/oauth/',
+  })
+    .then((res) => {
+      localStorage.setItem('token', res.data.user.token);
+      dispatch(createLoginSuccess());
+      dispatch(ShowModal({ modalShow: false }));
+    })
+    .catch((error) => {
+      dispatch(createLoginFailure(error));
+    });
+};
+
+export const googleLoginAction = () => (dispatch) => {
+  const provider = new firebase.auth.GoogleAuthProvider();
+  return firebase.auth().signInWithPopup(provider)
+    .then((result) => {
+      const token = result.credential.accessToken;
+      // dispatch method to remove current error
+      dispatch(socialLoginAction(token, 'google'));
+    }).catch(() => {
+      // dispatch method to raise error with name of the provider.
```

Comment:
Is there a missing dispatch here? I think it's best to always dispatch an action to the store whenever a request fails

Reply:
You're right, there is meant to be one there. No implemented yet. Thank you.